### PR TITLE
prototype: yaml-to-OSCAL transformer

### PR DESCRIPTION
**Pull request by degenaro**
Contribution is to help kubernetes produce NIST OSCAL, comprising: a folder with README and diagrams, a requirements spread sheet that documents how kubernetes results yaml fields are  mapped by the transformer to OSCAL json, a python trestle-based code module that realizes yaml-to-oscal transformations, a Makefile to install prereqs and run the transformer, and OSCAL assessment-results samples.

### CODE REVIEW

**Comment by sunstonesecure-robert** [comment]
overall looks great.  I think we just need to discuss the administrative errata about how we are going to be custodians of this peripheral to the main Trestle project.

and then a minor nit about whether we want to explicitly indicate in the path the trestle origination, ie instead of oscal-converter, maybe trestle-oscal-converter or just trestle-converter, following similar conventions as *-adapter?

**Comment by degenaro** [comment]
Thx for the comments. With respect to the "minor nit", I'll add my $.02. The present code takes YAML produced by Kube and produces OSCAL as JSON. It is properly called an oscal-transformer or kube-to-oscal-transformer or kube-yaml-to-oscal-assessment-results-json-transformer or such. The fact the the code takes advantage of the trestle SDK is nice, but not critical to the job at hand. We could image writing, with more difficulty, a GOLANG (or Fortran or Java or...) version that did not rely on trestle, and it too would be an oscal-transformer. That said, if encoding the dependency on trestle in the name is desirable I'm not opposed.

File: policy-report/oscal-transformer/Makefile
Comment (sunstonesecure-robert) [inline policy-report/oscal-transformer/Makefile:3]: we will need to remove and modify for CNCF compatibility

**Review by JimBugwadia — CHANGES_REQUESTED** [review]

File: policy-report/oscal-transformer/Makefile
Comment (degenaro) [reply policy-report/oscal-transformer/Makefile:3]: Jim, is anonymity the standard methodology used in this project? In all the open source projects I've contributed to, nominally the artifact donator gives license (such as Apache 2) for use. This also provides provenance for those so concerned.

Comment (degenaro) [reply policy-report/oscal-transformer/Makefile:3]: If I read this https://github.com/cncf/k8s-conformance/blob/master/terms-conditions/Certified_Kubernetes_Terms.md Participant Marks section correctly it seems to confirm my `gives license` assertion above, no?

Comment (JimBugwadia) [reply policy-report/oscal-transformer/Makefile:3]: @degenaro - good questions. 

I don't think the certification guidelines cover licensing.  I was going by the source code copyrights I see in other SIG projects. 

Workgroups are not supposed to publish code, and hence the goal would be for us to eventually migrate code we prototype and mature to a SIG repository.

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/eks/eks.go

https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/cmd/manager/main.go

We can ask the CNCF and Kubernetes maintainers for more guidance.

Comment (JimBugwadia) [reply policy-report/oscal-transformer/Makefile:3]: Posted a question at: https://kubernetes.slack.com/archives/CD6LAC15M/p1656376782867229

Comment (JimBugwadia) [reply policy-report/oscal-transformer/Makefile:3]: @degenaro - here was the response:

> https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#rules-for-new-repositories and https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#rules-for-donated-repositories are probably good starting points. Ideally, checking all those boxes for contributions/licenses/etc and making sure any additions/contributions to the repo keep those boxes checked.

Comment (degenaro) [reply policy-report/oscal-transformer/Makefile:3]: According to this `Additions of the standard Kubernetes header to code created by the contributors can occur post-transfer, but should ideally occur shortly thereafter.` the Copyright should remain, is how I read instructions.

Comment (degenaro) [reply policy-report/oscal-transformer/Makefile]: I claim to have fixed the copyright issue in accordance with the boilerplate here https://github.com/kubernetes/kubernetes/blob/master/hack/boilerplate/boilerplate.go.txt.

Comment (JimBugwadia) [reply policy-report/oscal-transformer/Makefile]: Hi @degenaro - thanks! I see you updated to `Copyright 2022 Lou DeGenaro (degenaro@us.ibm.com)`.

My suggestion would be to either remove the line for now, or change to `Copyright 2022 The CNCF Policy Working Group Authors`.  If we propose the project for SIG adoption, we can update to `Copyright 2022 The Kubernetes Authors.` at that time, to match the referenced boilerplate.

@sunstonesecure-robert - what are your thoughts?

**Review by JimBugwadia — APPROVED** [review]

File: policy-report/oscal-transformer/Makefile
Comment (degenaro) [reply policy-report/oscal-transformer/Makefile]: Hey @JimBugwadia, I changed the copyright per your suggestion. Thanks for the approval!

Comment (degenaro) [reply policy-report/oscal-transformer/Makefile]: `Needs lgtm label` it seems before merge is permitted.

**Pull request merged**
